src/Components/Components/src/Auth/AuthorizeRouteView.cs

```diff
@@ -18,6 +18,21 @@ namespace Microsoft.AspNetCore.Components
     /// </summary>
     public sealed class AuthorizeRouteView : RouteView
     {
+        private readonly RenderFragment<AuthenticationState> _renderAuthorizedDelegate;
+        private readonly RenderFragment<AuthenticationState> _renderNotAuthorizedDelegate;
+        private readonly RenderFragment _renderAuthorizingDelegate;
+
+        public AuthorizeRouteView()
+        {
+            // Cache the rendering delegates so that we only construct new closure instances
+            // when they are actually used (e.g., we never prepare a RenderFragment bound to
+            // the Authorizing content except when you are displaying that particular state)
+            var renderBaseRouteViewDelegate = (RenderFragment)base.Render;
```

Comment:
I'm not sure you need this, you could inline base.Render into that delegate. It's a super nitpick, but would remove 1 (count em, 1!) allocation

Reply:
Changed it to this:

```cs
RenderFragment renderBaseRouteViewDelegate = builder => base.Render(builder);
_renderAuthorizedDelegate = authenticateState => renderBaseRouteViewDelegate;
```

Not sure if that's what you meant. Whichever way this is done it seems like it has to allocate at least 2 delegates - one for the `RenderFragment<T>` and another for the `RenderFragment` that it returns.